src/unittests/f32_interval.spec.ts

```diff
@@ -3203,3 +3204,43 @@ g.test('faceForwardIntervals')
       `faceForwardInterval([${x}], [${y}], [${z}]) returned [${got}]. Expected [${expected}]`
     );
   });
+
+interface RefractCase {
+  input: [number[], number[], number];
+  expected: IntervalBounds[];
+}
+
+g.test('refractInterval')
+  .paramsSubcasesOnly<RefractCase>(
+    // prettier-ignore
+    [
+      // k < 0
+      { input: [[1, 1], [0.1, 0], 10], expected: [[0], [0]] },
+
+      // k contains 0
+      { input: [[1, 1], [0.1, 0], 1.005038], expected: [kAny, kAny] },
+
+      // k > 0
+      // vec2
+      { input: [[1, 1], [1, 0], 1], expected: [[hexToF64(0xbff00000, 0xc0000000), hexToF64(0xbfefffff, 0x40000000)], [1]] }, // ~[-1, 1]
```

Comment:
named constants would be nice here.

Reply:
I rolled the common -1 bounds up into a constant. I have left the others as values, since they are one off usages, but reformatted the lines to be more readable.